test/infrastructure/inmemory/pkg/server/api/watch.go

```diff
@@ -108,6 +109,35 @@ func (h *apiServerHandler) watchForResource(req *restful.Request, resp *restful.
 		return err
 	}
 
+	if resourceVersion != "" {
+		parsedResourceVersion, err := strconv.ParseUint(resourceVersion, 10, 64)
+		if err != nil {
+			return err
+		}
+		// Gets at client to the resource group.
+		inmemoryClient := h.manager.GetResourceGroup(resourceGroup).GetClient()
+
+		list, err := h.apiV1list(ctx, req, gvk, inmemoryClient)
+		if err != nil {
+			return err
+		}
+		var objResourceVersion uint64
+		for _, obj := range list.Items {
+			objResourceVersion, err = strconv.ParseUint(obj.GetResourceVersion(), 10, 64)
+			if err != nil {
+				return err
+			}
+			if objResourceVersion <= parsedResourceVersion {
```

Comment:
Todo: make sure we don't need `<`

```suggestion
			if objResourceVersion < parsedResourceVersion {
```

Reply:
I think it's correct based on the docs https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes

But good to verify